sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -456,6 +458,10 @@ class AccountState(
 
     ethTx.semanticValidity()
 
+    if (!params.allowUnprotectedTxs && tx.isLegacy && !tx.isEIP155) {
+      throw new IllegalArgumentException("Legacy unprotected transaction are not allowed.")
+    }
```

Comment:
we need to keep this check in the `AccountSidechainNodeViewHolder.applyLocallyGeneratedTransactions` method, since it is not a consensus rule.
Legacy unprotected transactions are allowed to be a part of the block.

Reply:
Another reason to move this check in `AccountSidechainNodeViewHolder.applyLocallyGeneratedTransactions`is that `validate` is called also for remotely generated txs, so legacy txs created by nodes that allow them would be rejected.